win32/src/winapi/ucrtbase.rs

```diff
@@ -6,12 +6,37 @@ use crate::Machine;
 const TRACE_CONTEXT: &'static str = "ucrtbase";
 
 #[win32_derive::dllexport(cdecl)]
-pub fn _initterm(_machine: &mut Machine, start: u32, end: u32) -> u32 {
+pub async fn _initterm(machine: &mut Machine, start: u32, end: u32) -> u32 {
+    if (end - start) % 4 != 0 {
+        panic!("unaligned _initterm");
+    }
+    let slice = machine.mem().sub(start, end - start).as_slice_todo();
+    let slice =
+        unsafe { std::slice::from_raw_parts(slice.as_ptr() as *const u32, slice.len() / 4) };
+    for &addr in slice {
+        if addr != 0 {
+            machine.call_x86(addr, vec![]).await;
+        }
+    }
     0
 }
 
 #[win32_derive::dllexport(cdecl)]
-pub fn _initterm_e(_machine: &mut Machine, start: u32, end: u32) -> u32 {
+pub async fn _initterm_e(machine: &mut Machine, start: u32, end: u32) -> u32 {
+    if (end - start) % 4 != 0 {
+        panic!("unaligned _initterm_e");
+    }
+    let slice = machine.mem().sub(start, end - start).as_slice_todo();
+    let slice =
+        unsafe { std::slice::from_raw_parts(slice.as_ptr() as *const u32, slice.len() / 4) };
+    for &addr in slice {
+        if addr != 0 {
+            let err = machine.call_x86(addr, vec![]).await;
+            if err != 0 {
```

Comment:
Is it intentional this doesn't match initterm?

I looked at the win2k msvcrt.dll I had but it only has initterm, no _e variant, and its behavior matches the function you wrote.

Reply:
Yes, `_initterm` calls `PVFV` (`void (*)(void)`) functions, whereas `_initterm_e` calls `PIFV` (`int (*)(void)`) functions, and the latter short circuits on a non-zero return value.

Reply:
(It's totally possible this behavior has changed over time though)